Fix:
directive.c:1212:20: error: implicitly declaring library function 'stpcpy' with type 'char *(char *, const char *)' [-Werror,-Wimplicit-function-declaration]
main.c:779:23: error: implicitly declaring library function 'stpcpy' with type 'char *(char *, const char *)' [-Werror,-Wimplicit-function-declaration]
system.c:2510:20: error: implicit declaration of function 'readlink' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
system.c:2623:15: error: implicitly declaring library function 'stpcpy' with type 'char *(char *, const char *)' [-Werror,-Wimplicit-function-declaration]
expand.c:427:16: error: implicitly declaring library function 'stpcpy' with type 'char *(char *, const char *)' [-Werror,-Wimplicit-function-declaration]
support.c:530:19: error: implicitly declaring library function 'stpcpy' with type 'char *(char *, const char *)' [-Werror,-Wimplicit-function-declaration]
diff -ru src/directive.c src/directive.c
--- src/directive.c.orig	2008-04-03 10:06:09.000000000 -0500
+++ src/directive.c	2021-02-16 01:44:24.000000000 -0600
@@ -35,6 +35,7 @@
  * are placed here.
  */
 
+#include <string.h>
 #if PREPROCESSED
 #include    "mcpp.H"
 #else
diff -ru src/expand.c src/expand.c
--- src/expand.c.orig	2008-03-18 03:50:41.000000000 -0500
+++ src/expand.c	2021-02-16 01:44:33.000000000 -0600
@@ -34,6 +34,7 @@
  * The macro expansion routines are placed here.
  */
 
+#include <string.h>
 #if PREPROCESSED
 #include    "mcpp.H"
 #else
diff -ru src/main.c src/main.c
--- src/main.c.orig	2021-02-15 21:23:30.000000000 -0600
+++ src/main.c	2021-02-16 01:44:43.000000000 -0600
@@ -35,6 +35,7 @@
  * The post-preprocessing routines are also placed here.
  */
 
+#include <string.h>
 #if PREPROCESSED                /* Use "pre-preprocessed" header    */
 #include    "mcpp.H"
 #else
diff -ru src/support.c src/support.c
--- src/support.c.orig	2021-02-15 21:23:30.000000000 -0600
+++ src/support.c	2021-02-16 01:44:29.000000000 -0600
@@ -70,6 +70,7 @@
  *              buffer.
  */
 
+#include <string.h>
 #if PREPROCESSED
 #include    "mcpp.H"
 #else
diff -ru src/system.c src/system.c
--- src/system.c.orig	2021-02-15 21:23:30.000000000 -0600
+++ src/system.c	2021-02-16 01:44:37.000000000 -0600
@@ -36,6 +36,7 @@
  *      1. specify the constants in "configed.H" or "noconfig.H",
  *      2. append the system-dependent routines in this file.
  */
+#include <string.h>
 #if PREPROCESSED
 #include    "mcpp.H"
 #else
